### PR TITLE
add rule for file name style

### PR DESCRIPTION
**Pull request by davidjgoss**
Sticking to a set of known styles at first, could be enhanced to support a custom pattern as well.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -85,6 +86,23 @@ will turn on the `no-unnamed-features` rule.
...
+
+```json
+{
+  "file-name": ["on", {"style": "TitleCase"}]
+}
+```
+
+The list of supported styles is:
+
+- `TitleCase`
```
Comment (vsiakka) [inline README.md:101]: I think that `TitleCase` refers to strings where each word starts with a capital letter and spaces between words are maintained. 
Sources: 
- https://wiki.c2.com/?CapitalizationRules
- https://en.wikipedia.org/wiki/Camel_case (3d paragraph)
`Title Case` maps exactly to the output of `_.startCase()`


When describing strings where each word starts with a capital letter and there are no spaces between words, the term I'm familiar with is `PascalCase`. 

Would you mind using that terminology instead? I think we could support both `Title Case` and `PascalCase`

File: README.md [hunk 2]
```diff
@@ -85,6 +86,23 @@ will turn on the `no-unnamed-features` rule.
...
+- `camelCase`
+- `kebab-case`
+- `snake_case`
+
```
Comment (vsiakka) [inline README.md:105]: Do you mind adding a short description of what each style means?

**Comment by vsiakka** [comment]
Thanks for the PR @davidjgoss ! I added a couple of comments, please take a look and let me know if the changes I'm asking for sound reasonable to you :)

File: README.md (hunk 1 above)
Comment (davidjgoss) [reply README.md]: Okay makes sense - done

File: README.md (hunk 2 above)
Comment (davidjgoss) [reply README.md:106]: Sure - done

**Comment by vsiakka** [comment]
Thanks for making the changes!
Published with v4.1.0

**Pull request merged**